src/java/io/bazel/rulesscala/scalac/ScalacInvoker3.java

```diff
@@ -1,45 +1,34 @@
 package io.bazel.rulesscala.scalac;
 
+import dotty.tools.dotc.Driver;
 import io.bazel.rulesscala.scalac.compileoptions.CompileOptions;
-import java.nio.file.Paths;
-import java.nio.file.Files;
-import scala.Tuple2;
+import io.bazel.rulesscala.scalac.reporter.ProtoReporter;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
-import dotty.tools.dotc.reporting.Reporter;
-import dotty.tools.dotc.Compiler;
-import dotty.tools.dotc.Driver;
-import dotty.tools.dotc.core.Contexts;
-import dotty.tools.io.AbstractFile;
-
-//Invokes Scala 3 compiler
-class ScalacInvoker{
+// Invokes Scala 3 compiler
+class ScalacInvoker {
```

Comment:
PR is draft since it lacks tests.
Any advice on how to setup diagnostic tests for Scala 3 is very welcome.